HLDB/Lib/DBModel.swift

```diff
@@ -10,20 +10,20 @@ import Foundation
 import Result
 import BrightFutures
 
-public class DBModel {
+public class DBModel <BackDB: AbstractDB, BackDBQueue: AbstractDBQueue where BackDB.Cursor : LazySequenceType, BackDB.Cursor.Generator.Element == NSDictionary, BackDBQueue.DB == BackDB> {
   public let dbFileName = "hldb"
-  public let db: HLDB.DB
-  public var tables = Dictionary<String, HLDB.Table>()
+  public let db: DB<BackDB, BackDBQueue>
+  public var tables = Dictionary<String, Table<BackDB, BackDBQueue>>()
```

Comment:
maybe it makes sense to have a `typealias BackTable = Table<BackDB, BackDBQueue>>` so the rest of the file is neater